src/tactic/lint.lean

```diff
@@ -690,22 +701,24 @@ format.intercalate "\n\n" <$> results.to_list.reverse.mmap (λ ⟨fn, results⟩
 
   Returns a `name_set` containing the names of all declarations that fail any check in `check`,
   and a `format` object describing the failures. -/
-meta def lint_aux (l : list declaration) (group_by_filename : option nat)
+meta def lint_aux (decls : list declaration) (group_by_filename : option nat)
     (where_desc : string) (slow verbose : bool) (checks : list (name × linter)) :
   tactic (name_set × format) := do
-results ← lint_core l checks,
+e ← get_env,
+let non_auto_decls := decls.filter (λ d, ¬ d.to_name.is_internal ∧ ¬ d.is_auto_generated e),
```

Comment:
I think this line should go into `lint_core`.  The `lint_core` function is called directly by the `mk_nolint.lean` script, which would otherwise need to duplicate this functionality.